src/codemodder/logging.py

```diff
@@ -13,6 +16,28 @@ class OutputFormat(Enum):
     HUMAN = "human"
     JSON = "json"
 
+    def __str__(self):
+        """For rendering properly in argparse help."""
+        return self.value.lower()
+
+
+class CodemodderJsonFormatter(jsonlogger.JsonFormatter):
+    project_name: Optional[str]
+
+    def __init__(self, *args, project_name: Optional[str] = None, **kwargs):
+        self.project_name = project_name
+        super().__init__(*args, **kwargs)
+
+    def add_fields(self, log_record, record, message_dict):
```

Comment:
I'm tempted to ignore this block for the purposes of coverage. I tried to write some tests here but it's proving fairly difficult to get the mocks right.

Reply:
I'm super surprised codecov says most of this isn't tested. Curious, if you put a debugger in here and run codemodder as pixeebot would, it does it it, right? That would tell me integration tests would test this but right now we only calc coverage based on unit tests. This makes sense bc the way we run integration tests with subprocess.run creates a separate process and pytest / cov can't calculate coverage there.

Is there any help from the python-json-logger docs in regards to testing? Does chatgpt help in any way here? If you've tried these avenues and felt like you've spent sufficient time, I would be ok with adding no-cover here though it does break my heart a little.

Reply:
This codepath is only covered if we use `--log-format=json` which probably doesn't happen in any of the integration tests right now. This basically just boils down to level of effort vs reward right now and I can't afford to spend any more time trying to get this covered.